pkg/virtualKubelet/translation/mutation.go

```diff
@@ -180,3 +187,52 @@ func ChangePodIp(newPodCidr string, oldPodIp string) (newPodIp string) {
 	}
 	return strings.TrimSuffix(newPodIpBuilder.String(), ".")
 }
+
+func TranslateSA(destPod *v1.Pod, origPod *v1.Pod, remoteSecrets []interface{}) (*v1.Pod, error) {
+	var secret *v1.Secret = nil
+	for _, obj := range remoteSecrets {
+		s, ok := obj.(*v1.Secret)
+		if !ok {
+			continue
+		}
+		if sa, ok := s.Labels["kubernetes.io/service-account.name"]; ok && sa == origPod.Spec.ServiceAccountName {
+			secret = s
+			break
+		}
+	}
+	if secret == nil {
+		return nil, kerrors.NewServiceUnavailable("service account token not found")
```

Comment:
I do not know if this error suits our case. I would use either `notFound` or a custom error